ledger/ledger-api-integration-tests/src/test/itsuite/scala/com/digitalasset/platform/tests/integration/ledger/api/PartyManagementServiceIT.scala

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.platform.tests.integration.ledger.api
+
+import com.digitalasset.ledger.api.testing.utils.{
+  AkkaBeforeAndAfterAll,
+  SuiteResourceManagementAroundAll
+}
+import com.digitalasset.ledger.api.v1.admin.party_management_service.PartyManagementServiceGrpc.PartyManagementService
+import com.digitalasset.ledger.client.services.admin.PartyManagementClient
+import com.digitalasset.platform.apitesting.MultiLedgerFixture
+import com.digitalasset.platform.esf.TestExecutionSequencerFactory
+import org.scalatest._
+import org.scalatest.concurrent.AsyncTimeLimitedTests
+import org.scalatest.time.Span
+import org.scalatest.time.SpanSugar._
+import scalaz.syntax.tag._
+
+@SuppressWarnings(
+  Array(
+    "org.wartremover.warts.Any",
+    "org.wartremover.warts.Option2Iterable",
+    "org.wartremover.warts.StringPlusAny"
+  ))
+class PartyManagementServiceIT
+    extends AsyncWordSpec
+    with AkkaBeforeAndAfterAll
+    with MultiLedgerFixture
+    with SuiteResourceManagementAroundAll
+    with AsyncTimeLimitedTests
+    with TestExecutionSequencerFactory
+    with Matchers
+    with Inside
+    with OptionValues {
+
+  override def timeLimit: Span = 15.seconds
+
+  override protected def config: Config = Config.default
+
+  private def partyManagementClient(stub: PartyManagementService): PartyManagementClient = {
+    new PartyManagementClient(stub)
+  }
+
+  "Party Management Service" when {
+
+    "returning the participant ID" should {
+      "succeed" in allFixtures { c =>
+        partyManagementClient(c.partyManagementService)
+          .getParticipantId()
+          .map(id => id.unwrap.isEmpty shouldBe false)
+
+      }
+    }
+
+    "allocating parties" should {
+      "list the new party" in allFixtures { c =>
+        val client = partyManagementClient(c.partyManagementService)
+
+        for {
+          initialParties <- client.listKnownParties()
+          result <- client.allocateParty(Some("party"), Some("hint"))
+          finalParties <- client.listKnownParties()
+        } yield {
+          initialParties.length shouldBe 0
```

Comment:
This assertion will not hold when running the test against a long running ledger (as in DAML-on-X). Maybe the only assertion you can make is that the party is not contained in `initialParties` but IS contained in `finalParties`.

Reply:
Good point, this test is not very good. And in addition to the above, it also needs to be able to deal with ledgers that reject the call to allocate a party.